File: 938-range-sum-of-bst/938-range-sum-of-bst.rs
// Definition for a binary tree node.
// #[derive(Debug, PartialEq, Eq)]
// pub struct TreeNode {
//   pub val: i32,
//   pub left: Option<Rc<RefCell<TreeNode>>>,
//   pub right: Option<Rc<RefCell<TreeNode>>>,
// }
// 
// impl TreeNode {
//   #[inline]
//   pub fn new(val: i32) -> Self {
//     TreeNode {
//       val,
//       left: None,
//       right: None
//     }
//   }
// }
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn range_sum_bst(root: Option<Rc<RefCell<TreeNode>>>, low: i32, high: i32) -> i32 {
        let r: i32 = match root {
            None => 0,
            Some(node) => {
                let mut res: i32 = 0;
                let node = node.borrow();
                if ((node.val >= low) & (node.val <= high)) {
                    res += node.val;
                }
                res += Solution::range_sum_bst(node.left.clone(), low, high);
                res += Solution::range_sum_bst(node.right.clone(), low, high);
                return res;
                },
        };
        return r;
    }
}

File: 938-range-sum-of-bst/938-range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        if not(root):
            return 0
        r = 0
        if (root.val >= low) and (root.val <= high):
            r += root.val
        r += self.rangeSumBST(root.left, low, high)
        r += self.rangeSumBST(root.right, low, high)
        return r